lib/src/rules/avoid_function_literals_in_foreach_calls.dart

```diff
@@ -31,6 +33,27 @@ people.forEach(print);
 ```
 ''';
 
+bool _hasMethodChaining(MethodInvocation node) {
+  var exp = node.target;
+  while (exp is PrefixedIdentifier ||
+      exp is MethodInvocation ||
+      exp is PropertyAccess) {
+    if (exp is PrefixedIdentifier) {
+      exp = exp.prefix;
+    } else if (exp is MethodInvocation) {
+      return true;
+    } else if (exp is PropertyAccess) {
+      exp = exp.target;
+    }
+  }
+  return false;
+}
+
+bool _isNonNullableIterable(DartType? type) =>
+    type != null &&
+    type.nullabilitySuffix != NullabilitySuffix.question &&
```

Comment:
I'm not sure how this works for `dynamic`. Consider
```
void f(dynamic iter) => iter?.forEach(...);
```
Is the suffix always a question even when it isn't explicitly given?

I'll just note that an equivalent test might be to look for the `?.` operator on the method invocation.

Reply:
> Is the suffix always a question even when it isn't explicitly given?

I **_believe_** so?  (@scheglov may knot for sure.)

In any event, I added a `dynamic` test case and it seems to work as written.

Thanks for the catch!

Reply:
The type `dynamic` is always nullable, and it does not have any `nullabilitySuffix`, it is always `NullabilitySuffix.none`. Note that the suffix is not the same as nullability, e.g. `FutureOr<int?>` is nullable.